torch/testing/_internal/common_nn.py

```diff
@@ -4891,7 +4906,10 @@ def noncontiguize(self, obj):
             if tensor.size(d) > 1:
                 dim = d + 1
                 break
-        noncontig = torch.stack([torch.empty_like(tensor), tensor], dim).select(dim, 1).detach()
+
+        # Note: hstack is more concise here but doesn't support complex
```

Comment:
I don't think that's true. hstack is supported for complex on cpu (as shown below) and also on cuda (I know this since I very recently added autograd test for it)
```
>>> a=torch.randn(3, dtype=torch.cfloat)
>>> torch.hstack((a, a))
tensor([-0.1724+0.7514j, -0.0378-0.2850j, -0.4960+0.8649j, -0.1724+0.7514j,
        -0.0378-0.2850j, -0.4960+0.8649j])
```

Reply:
I'm assuming the comment means stack and not hstack

Reply:
Yep, unfortunately my comment is still inaccurate since technically both stack/hstack support complex. Its just that they cannot be automatically differentiated for complex - though that is something anjali is working on and is landing soon. So the final version of this diff will probably end up still using stack